pages/admin/treatments/index.tsx

```diff
@@ -0,0 +1,293 @@
+            TreatmentNotification.deleteSuccess();
+            await refetch();
+            setModalState({
+                ...modalState,
+                isOpen: false,
+            });
+        } catch (error) {
+            console.error('Error al eliminar el tratamiento', error);
+            TreatmentNotification.deleteError();
+        } finally {
+            setIsSubmitting(false);
+        }
+    }
+
+    function handleEditTreatment(treatment: Treatment) {
+        setModalState({
+            isOpen: true,
+            mode: 'edit',
+            selectTreatment: treatment,
+        });
+    }
+
+    function handleAddTreatment() {
+        if (!treatments) return;
+        setModalState({
+            isOpen: true,
+            mode: 'create',
+        });
+    }
+    function handleDeleteTreatment(treatment: Treatment) {
+        setModalState({
+            isOpen: true,
+            mode: 'delete',
+            selectTreatment: treatment,
+        });
+    }
+
+    const handleOnSubmit = async (
+        treatment?: Omit<Treatment, 'id'>,
+    ): Promise<void> => {
+        if (modalState.mode === 'delete') {
+            await handleConfirmDelete();
+            return;
+        }
+        if (modalState.mode === 'edit') {
+            await handleUpdateTreatment(treatment);
+            return;
+        }
+        await handleCreateTreatment(treatment);
+    };
+
+    return (
+        <>
+            <Box
+                p={isSmallScreen ? 3 : 8}
+                m={isSmallScreen ? 3 : 8}
+                bg='white'
+                borderRadius='lg'
+                className='light'
+            >
+                <Box
+                    display='flex'
+                    justifyContent='space-between'
+                    alignItems='center'
+                    mb={6}
+                >
+                    <Text fontSize='2xl' fontWeight='bold'>
+                        Tratamientos
+                    </Text>
+                    <Button
+                        _hover={{ bg: 'blue.500' }}
+                        p={2}
+                        size='lg'
+                        fontWeight='bold'
+                        colorPalette='blue'
+                        onClick={handleAddTreatment}
+                    >
+                        <Plus size={20} />
+                        Crear Tratamiento
+                    </Button>
+                </Box>
+
+                <SearchFilters
+                    filters={filters}
+                    onFilterChange={handleFilterChange}
+                    onSortChange={handleSortChange}
+                    showSort={true}
+                />
+
+                <AdminTable
+                    data={filteredTreatments}
+                    columns={tableColumns}
+                    isLoading={loading}
+                />
+            </Box>
+            <TreatmentModal
+                isOpen={modalState.isOpen}
```

Comment:
💭 Properties such as isOpen and mode should be handled by the TreatmentModal since is logic that is not responsibility of this index

Similar with the `onTreatmentCreated` that prop sounds that can be handled in the obSubmit 🤔